### PR TITLE
Boat Class

### PR DESCRIPTION
**Pull request by ArupUkil**
Added Boat class, made a test for the Boat class, and added two Boats to the traders inventory.

### CODE REVIEW

File: src/main/java/Boat.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        this.maxSpeed += 2;
+    }
+
+    @Override
+    public void downgradeSpeed() {
+        if (this.maxSpeed >= 0){
+            this.maxSpeed -= 2;
+        }
+    }
+
```
Comment (Maddobun) [inline src/main/java/Boat.java:20]: include method return documentations

File: src/main/java/Boat.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+public class Boat implements Tradable, Drivable {
+    private int maxSpeed;
+
```
Comment (Maddobun) [inline src/main/java/Boat.java:4]: Include documentations.

**Review by Maddobun — COMMENTED** [review]
require minor cleanup

**Comment by ArupUkil** [comment]
> require minor cleanup

You're correct but you missed my intentional problem which is that I had downgradeSpeed has a condition that makes sure that maxSpeed doesn't go below 0 but it fails because maxSpeed is >= 0 not > 0 so negative maxSpeed is possible. Also need to test that the maxSpeed doesn't go below 0.

**Pull request merged**